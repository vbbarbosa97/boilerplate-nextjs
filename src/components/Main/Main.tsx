import * as S from './styles/Main.styles';

interface MainProps {
	title?: string;
	description?: string;
}

export const Main = ({
	title = 'React Avançado',
	description = 'Typescript, ReactJS, NExtJS e Styled Components',
}: MainProps) => {
	return (
		<S.Wrapper>
			<S.Logo src="/img/logo.svg" alt="Imagem de um átomo e React Avançado escrito ao lado." />
			<S.Title>{title}</S.Title>
			<S.Description>{description}</S.Description>
			<S.Illustration src="/img/hero-illustration.svg" alt="Um desenvolvedor de frente para uma tela com código." />
		</S.Wrapper>
	);
};
