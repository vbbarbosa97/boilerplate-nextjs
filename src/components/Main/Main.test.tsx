import { render, screen } from '@testing-library/react';
import { Main } from './Main';

describe('<Main />', () => {
	it('deve_renderizar_o_heading', () => {
		const { container } = render(<Main />);

		expect(screen.getByRole('heading', { name: /react avanÃ§ado/i }));
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render the colors correctly', () => {
		const { container } = render(<Main />);

		expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' });
	});
});
