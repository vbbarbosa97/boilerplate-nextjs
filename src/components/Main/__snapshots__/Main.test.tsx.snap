// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> deve_renderizar_o_heading 1`] = `
<main>
  <h2>
    React Avan√ßado
  </h2>
</main>
`;
